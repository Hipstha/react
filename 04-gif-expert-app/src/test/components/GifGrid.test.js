import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

  const category = 'Pokemon';
  
  test('Debe cargar <GifGrid /> correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    }); // crea mock data en el fetch
    const wrapper = shallow( <GifGrid category={category} /> );
    expect( wrapper ).toMatchSnapshot();
  });

  test('Debe de mostrar items cuando se cargan imÃ¡genes con useFetchGifs', () => {
    const gifs = [{
      id: 'abc',
      url: 'https://localhost/culauqier/cosa.jpg',
      title: 'cualquier cosa'
    }, {
      id: '123',
      url: 'https://localhost/culauqier/cosa.jpg',
      title: 'cualquier cosa'
    }];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    }); // crea mock data en el fetch
    const wrapper = shallow( <GifGrid category={category} /> );

    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('p').exists() ).toBe(false);
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );


  });

});