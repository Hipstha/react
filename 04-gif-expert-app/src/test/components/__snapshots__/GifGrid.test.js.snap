// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe cargar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3>
    Pokemon
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe de mostrar items cuando se cargan im√°genes con useFetchGifs 1`] = `
<Fragment>
  <h3>
    Pokemon
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="cualquier cosa"
      url="https://localhost/culauqier/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="cualquier cosa"
      url="https://localhost/culauqier/cosa.jpg"
    />
  </div>
</Fragment>
`;
